src/Propel/Runtime/ActiveQuery/ModelCriteria.php

```diff
@@ -540,7 +547,7 @@ public function getSelect()
      *
      * @return \Propel\Runtime\ActiveQuery\Join the previousJoin for this ModelCriteria
      */
-    public function getPreviousJoin()
+    public function getPreviousJoin(): Join
```

Comment:
Do we agree this should be `?Join` as `useQuery` may not be called yet?

Reply:
Many of those might have to be nullable as doc blocks where most likely lying.